src/pages/SearchResults/index.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { useLocation } from "react-router";
+
+import EpisodeCard from "../../components/EpisodeCard";
+import TitleCount from "../../components/TitleCount";
+import CharacterList from "../../components/CharacterList";
+import { IEpisode, ISearchData } from "../../types/types";
+
+function SearchResults() {
+  const { state } = useLocation();
+
+  const episodeArray = (state as ISearchData)?.episodes?.results.map(
+    (episode: IEpisode) => {
+      return (
+        <div className="episode-list" key={episode.id}>
+          <EpisodeCard
+            id={episode.id}
+            season={episode.episode}
+            date={episode.air_date}
+            title={episode.name}
+            description="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore..."
+          />
+        </div>
+      );
+    }
+  );
+
+  return (
+    <>
+      {(state as ISearchData)?.episodes?.results?.length ||
+      (state as ISearchData)?.characters?.results?.length > 0 ? (
+        <div className="search-results-wrapper">
+          <div className="search-results-container">
+            <div className="search-characters">
+              <div className="search-results-title">Search Results</div>
+              <TitleCount
+                count={
+                  (state as ISearchData)?.characters?.info?.count > 0
+                    ? (state as ISearchData)?.characters?.info?.count
```

Comment:
Since you are using `(state as ISearchData)?.episodes?.results?.length` and `(state as ISearchData)?.characters?.info?.count` multiple times, you can define constants for them.